### ISSUE TITLE
Add support for internal links in previewHtml

### ISSUE THREAD

**Issue by tomi** [Microsoft/vscode#3676]
Hi,

At the moment the virtual documents created by TextDocumentContentProvider can contain links to external resources, but I would expect it to work with links that use the same scheme that was registered with registerTextDocumentContentProvider. Here's an example to show what I mean:

```javascript
"use strict";
var vscode = require('vscode');

function activate(context) {

    let initUri = vscode.Uri.parse('link-test://authority/page1');

    const provider = {
        provideTextDocumentContent: function(uri) {
            if (uri.path === "/page1") {
                return `<a href="link-test://authority/page2">Page 2</a>`;
            } else {
                return `<a href="link-test://authority/page1">Page 1</a>`;
            }
        }
    };

    let registration = vscode.workspace.registerTextDocumentContentProvider('link-test', provider);

    let disposable = vscode.commands.registerCommand('extension.testLinks', () => {
        return vscode.commands.executeCommand('vscode.previewHtml', initUri, vscode.ViewColumn.Two).then((success) => {
        }, (reason) => {
            vscode.window.showErrorMessage(reason);
        });

    });
    context.subscriptions.push(disposable, registration);
}
exports.activate = activate;
```

When I click the link Page 2 I would expect the provideTextDocumentContent callback to be called with the URI specified in the link.

**Comment by BMGburger** [comment]
It looks great and I'm impressed...Also hoping I'm posting this in the right place.

**Comment by jrieken** [comment]
I like

**Comment by jrieken** [comment]
strech for April - fyi @seanmcbreen

**Comment by jrieken** [comment]
Tasks

* [ ] support virtual document, file document, and commands
* [ ] support to jump to a line in a file document

**Comment by jrieken** [comment]
@tomi This is now implemented but be aware of how it works: In your case you have a virtual document that contains a link to another virtual document. Since the link only becomes _a link_ when rendered via the `vscode.previewHtml` command, your link must do the same. Therefore we have added support for _command-links_, like so `command:vscode.previewHtml?"link-test://auth/path"`. It tells VS Code to execute the command using the query-part of the URI.
